engines/advancedDetector.cpp

```diff
@@ -513,31 +503,56 @@ namespace Common {
 	DECLARE_SINGLETON(MD5CacheManager);
 }
 
-// Sync with engines/game.cpp
-static char flagsToMD5Prefix(uint32 flags) {
-	if (flags & ADGF_MACRESFORK) {
-		if (flags & ADGF_TAILMD5)
-			return 'e';
-		return 'm';
+
+static MD5Properties gameFileToMD5Props(const ADGameFileDescription *fileEntry, uint32 gameFlags) {
+	if(fileEntry && fileEntry->md5 && fileEntry->md5[0] != '\0'  && fileEntry->md5[1] == ':') {
+		switch (fileEntry->md5[0]) {
+		case 'r':
+			return kMD5MacResFork;
+		case 'x':
+			return (MD5Properties) (kMD5MacResFork | kMD5Tail);
+		case 'd':
+			return kMD5MacDataFork;
+		case 'z':
+			return (MD5Properties) (kMD5MacDataFork | kMD5Tail);
+		case 'f':
+			return kMD5Head;
+		case 't':
+			return kMD5Tail;
+		}
+	}
+
+	if(fileEntry->md5 && fileEntry->md5[0] == 'r'  && fileEntry->md5[0] == ':') {
+		return kMD5MacResFork;
+	}
+
+	MD5Properties ret = (MD5Properties) 0;
```

Comment:
We normally do not put space after type conversion, e.g. `(MD5Properties)0` and it is everywhere in this method. Also, instead of 0, I'd suggest putting something like 'kMD5Default' or 'kMD5Normal'

Reply:
Done